### ISSUE TITLE
Where to put type declaration file for JSX

### ISSUE THREAD

**Issue by ellisonbg** [jupyterlab/jupyterlab#2470]
I am working on the input/output collapsing and trying to use basic txs syntax in a few places. I have it working, but can't figure out where to put the `d.ts` file with the JSX type information. The code looks something like this:

```typescript
import {
    ElementAttrs, VirtualElement
} from '@phosphor/virtualdom';



/**
 * Define global element types to JSX namespace.
 */
declare global {
  namespace JSX {
    interface IntrinsicElements {
        ...
    }
    interface Element extends VirtualElement { }
  }
}
```

I have been successful in putting this code in two places:

1) In the actual `.tsx` file.
2) In a `jsx.ts` file that I `export * from...` in the `index.ts` of the module.

I have tried a million ways to put that in a separate `d.ts` file but it won't work. I put it into `typings/jsx/jsx.d.ts` and added a `///` reference in the `src/typings` file, but that won't pick it up. I need to use this file in multiple jlab packages, so having a quick and reusable way to handle it is desirable. I know there are broader issues around TSX and vdom stuff, but I don't want to solve them now, I just want to start using basic TSX syntax in a few places.

**Comment by blink1073** [comment]
I was able to get a minimal working example here: https://github.com/blink1073/jupyterlab/tree/tsx-example/packages/codemirror.  I made a `typings/tsx/tsx.d.ts`, referred to it from `src/typings.ts`, and used it in `src/test.tsx`, along with the changes to `tsconfig.json`.

**Comment by blink1073** [comment]
Relevant commit is https://github.com/blink1073/jupyterlab/commit/5293ad44bfeaac0248cbd3f2b46ce4c868b4ecd7.  The output is:

```
$ cat codemirror/lib/test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var virtualdom_1 = require("@phosphor/virtualdom");
virtualdom_1.h("button", null, "hi");
```

**Comment by ellisonbg** [comment]
Great, I got it to work, but did need to add the typing information to `all-packages` as well. But I think I have it all working in this PR #2474

**Comment by ellisonbg** [comment]
Part of the problem is that I tried working on this at midnight Friday night ;-)

**Comment by blink1073** [comment]
Fixed in #2474.
